docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+
+- You can run the benchmarks against [multiple Runtimes](./benchmarkdotnet.md#Multiple-Runtimes). It can be very useful when the regression has been introduced between .NET Core releases, for example: between netcoreapp2.2 and netcoreapp3.0.
+- You can run the benchmarks using provided [dotnet cli](./benchmarkdotnet.md#dotnet-cli). You can download few dotnet SDKs, unzip them and just run the benchmarks to spot the version that has introduced the regression to narrow down your investigation.
+- You can run the benchmarks using few [CoreRuns](./benchmarkdotnet.md#CoreRun). You can build the latest CoreFX in Release, create a copy of the folder with CoreRun and use git to checkout older commit. Then rebuild the CoreFX and run the benchmarks against the old and new builds. This can narrow down your investigation to the commit that has introduced the bug.
+
+### Confirmation
+
+When you identify and fix the regression, you should use [ResultsComparer](../src/tools/ResultsComparer/README.md) to confirm that you have solved the problem. Please remember that if the regression was found in a very common type like `Span<T>` and you are not sure which benchmarks to run, you can run all of them using `--filter *`.
+
+## Local CoreCLR Build
+
+Sometimes you might be willing to run the benchmarks using not only local CoreFX but also local CoreCLR build.
+
+Thanks to the simplicity of CoreRun, all you need to do is to copy all the runtimes files into the folder with CoreRun.
+
+But before you do it, you should copy/remeber the version of CoreCLR reported by BenchmarkDotNet:
+
+```log
+BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.17763.253 (1809/October2018Update/Redstone5)
+Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
+.NET Core SDK=3.0.100-preview-009812
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-SBBRNM : .NET Core ? (CoreCLR 4.6.27416.73, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+As you can see, before the "update" the version for the Job (not the Host process) was `4.6.27416.73`.
+
+The CoreFX build system has a built-in support for that, exposed by the [CoreCLROverridePath](https://github.com/dotnet/corefx/blob/0e7236fda21a07302b14030c82f79bb981c723a6/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits) build parameter:
+
+```cmd
+C:\Projects\corefx> build -Release /p:CoreCLROverridePath="C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release"
+```
+
+After the update, the reported CoreCLR version should be changed:
+
+```log
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-CYVJFZ : .NET Core ? (CoreCLR 4.6.27517.0, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+**It's very important to validate the reported version numbers** to make sure that you are running the benchmarks using the versions you think that you are using.
+
+## Benchmarking new API
```

Comment:
@stephentoub @ahsonkhan you both asked for that. I have tested that very carefully (I added new library to CoreFX, new type, new method to existing type) and everything works fine